internal/service/capsule/build.go

```
@@ -111,3 +112,18 @@ func (s *Service) validateImage(ctx context.Context, ref name.Reference) (string
 
 	return d.String(), nil
 }
+
+func (s *Service) getLookupDockerRef(ref name.Reference) name.Reference {
+	cfg := s.cfg.Cluster.DevRegistry
+	if !cfg.Enabled || cfg.Host == "" || cfg.ClusterHost == "" {
+		return ref
+	}
+
+	if ref.Context().RegistryStr() != cfg.Host {
+		return ref
+	}
+
+	r, _ := name.NewRegistry(cfg.ClusterHost, name.Insecure)
```

we should probably surface this error